### PR TITLE
Apis 1041 application view

### PR DESCRIPTION
**Pull request by RawToast**
Added basic application view to gatekeeper frontend.

### CODE REVIEW

File: app/assets/javascripts/developers.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ $(function(undefined) {
   (function() {
     var DEVELOPER_TABLE = 'developer-table';
     var dataTables = [
-      DEVELOPER_TABLE
+      DEVELOPER_TABLE, "applications-table"
```
Comment (atmd83) [inline app/assets/javascripts/developers.js]: we set the visual equivalent of a const for the developer table but not application table. we should be consistant, either: 
```
var DEVELOPER_TABLE = 'developer-table';
var APPLICATIONS_TABLE = 'applications-table';
```

or

```
var dataTables = ['developer-table', 'applications-table'];
```

File: app/model/Application.scala [hunk 2]
```diff
@@ -46,11 +46,32 @@ object ApplicationResponse {
...
+                                         subscriptionNames: Seq[String]) extends Application
+
+
+object SubscribedApplicationResponse {
+ implicit val format1 = Json.format[APIIdentifier]
+ implicit val formatRole = EnumJson.enumFormat(CollaboratorRole)
+ implicit val format2 = Json.format[Collaborator]
+ implicit val format3 = EnumJson.enumFormat(State)
+ implicit val format4 = Json.format[ApplicationState]
+ implicit val format5 = Json.format[SubscribedApplicationResponse]
```
Comment (jim-coffey) [inline app/model/Application.scala]: Indentation of implicits above

**Comment by jim-coffey** [comment]
Can I double just double check the naming of the applications connector endpoint `application/subscriptions` or `application/subscription/all` as per the services stub. I assume this is correct and the api-services-stub branch needs updating, but...?

**Comment by RawToast** [comment]
The endpoint is ```application/subscriptions``` there is a branch for the services stub with those changes.

File: app/assets/javascripts/developers.js [hunk 3]
```diff
@@ -42,14 +42,13 @@ $(function(undefined) {
...
-    var DEVELOPER_TABLE = 'developer-table';
     var dataTables = [
-      DEVELOPER_TABLE
+      'developer-table', 'applications-table'
     ];
 
     var getAllFilters = function (table) {
       return $('*[data-datatable-filter="' + table + '"]');
-    }
+    };
```
Comment (jim-coffey) [inline app/assets/javascripts/developers.js]: How dare you remove this semi-colon?

I know it was me with the hard coded strings in the first place, but I'd actually like to change this to : 

```
var dataTables = $('table.dataTable');
```

and the each loop at the bottom of the JS to read : 

```
$.each(dataTables, function (index, table) {
  var id = $(table).attr('id');
  var filters = getAllFilters(id);
  $.each(filters, dataTableEvents);
  $(table).DataTable();
});
```

...and of course remove the magic strings entirely.

**Comment by jim-coffey** [comment]
Go for it!

**Pull request merged**